File: Easy/JavaScript/942. DI String Match.js
var diStringMatch = function(s) {
    let n = s.length;
    let perm = Array(n + 1).fill(0);
    let start = 0;
    let end = n;

    for(let i = 0; i < s.length; i++){
        if(s.charAt(i) == 'I')
            perm[i] = start++;
        else
            perm[i] = end--;
    }
    perm[n] = start;

    return perm;
};

File: Easy/Python/942. DI String Match.py
class Solution(object):
    def diStringMatch(self, s):
        n = len(s)
        perm = []
        start = 0
        end = n

        for i in range(len(s)):
            if s[i] == 'I':
                perm.append(start)
                start += 1
            else:
                perm.append(end)
                end -= 1
        perm.append(start)

        return perm
